### PR TITLE
Fix autocorrect onChangeText firing (#22691)

### PR DESCRIPTION
**Pull request by ericlewis**
## Summary

This fixes an issue where autocorrect changes may cause `onChangeText` to not fire. It subscribes to the underlying `textStorage` of the UITextView. It is possible this fixes a few other issues, due to the nature of listening to the underlying textStorage changes. 

## Changelog

[iOS] [Fixed] - iOS firing onChangeText when choosing autocorrect options

## Test Plan

Open RNTester, and interact with textfields.

### CODE REVIEW

**Comment by ericlewis** [comment]
fixes #22691

**Comment by ericlewis** [comment]
Would appreciate some extra eyes on this, it appears to work, though.

**Comment by shergin** [comment]
cc @rigdern, @mandrigin 

That's really cool. I didn't know that the API that we use here exists! That also probably can help us to fix other Korean/Japanese/Chinese issues.

I will test it and hopefully land soon.

File: Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m [hunk 1]
```diff
@@ -150,11 +136,21 @@ - (instancetype)initWithTextView:(UITextView<RCTBackedTextInputViewProtocol> *)b
...
 
   return self;
 }
 
+#pragma mark -
+- (void)textStorage:(NSTextStorage *)textStorage
+  didProcessEditing:(__unused NSTextStorageEditActions)editedMask
+              range:(__unused NSRange)editedRange
+     changeInLength:(__unused NSInteger)delta {
+  [_backedTextInputView.textInputDelegate textInputDidChange];
```
Comment (shergin) [inline Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m]: I am worried a bit about over-firing. So, e.g. if only text attributes were changed, we don't want to send an event.

Comment (ericlewis) [reply Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m]: I will check if this is the case & fix it if that does occur.

**Comment by ericlewis** [comment]
@shergin I changed this PR- I originally removed code from the UITextField adapter implementation, this was a mistake I believe. So, listening to the NSTextStorage only applies to multiline TextInput components. I feel like it would make sense for us to use UITextView instead of UITextField for all TextInput's due to UITextView having lower level API access.

**Comment by ericlewis** [comment]
@zhongwuzw ah yes, I forgot to set the delegate in my cleaned up version. I will fix that. Good point on the remaining issues.

**Pull request closed without merging**